assayist/processor/main_analyzer.py

```diff
@@ -0,0 +1,135 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from assayist.processor.base import Analyzer
+from assayist.processor.logging import log
+
+
+class MainAnalyzer(Analyzer):
+    """
+    Look at the json brew output and write neomodels for the basic items identified.
+
+     * The Component
+     * The direct SourceLocation
+     * The Build itself
+     * Any produced Artifacts (RPMs or otherwise)
+     * Any RPMs used in the buildroot
+     * If it's an image build, an RPMs included in the image
+    """
+
+    buildroot_to_artifact = {}
+
+    def map_buildroot_to_artifact(self, buildroot_id, artifact_id):
+        """Store the mapping in self.buildroot_to_artifact."""
+        if buildroot_id in self.buildroot_to_artifact:
+            self.buildroot_to_artifact[buildroot_id].append(artifact_id)
+        else:
+            self.buildroot_to_artifact[buildroot_id] = [artifact_id]
+
+    def read_and_save_buildroots(self):
+        """Save and link the rpms used in the buildroot for each artifact."""
+        buildroots_info = self.read_metadata_file(self.BUILDROOT_FILE)
+        for buildroot_id in buildroots_info:
+            log.debug("Creating artifacts for buildroot %s", buildroot_id)
+            for rpm_info in buildroots_info[buildroot_id]:
+                rpm = self.get_or_create_rpm_artifact_from_hash(rpm_info)
+                if buildroot_id in self.buildroot_to_artifact:
+                    for artifact in self.buildroot_to_artifact[buildroot_id]:
+                        artifact.buildroot_artifacts.connect(rpm)
+
+    def construct_and_save_component(self, build_type, build_info):
+        """
+        Read the build info and contruct the Component.
+
+        Returns: (Component, canonical_version)
+
+        """
+        if build_type == 'build':  # rpm build
+            cnamespace = 'redhat'
+            cname = build_info['name']
+            ctype = 'rpm'
+            cversion = '%s-%s' % (build_info['version'], build_info['release'])
+        elif build_type == 'maven':
+            # What we really want is the contents of the "Maven groupId" etc fields.
+            # However they don't seem to be included in the brew response?!
```

Comment:
You have to make a second call to get them. I forgot about that when I was doing the original design. 

We'll need to call the equivalent of: `koji call --output-json getMavenBuild <build-id>`, which looks like it will return `build_id, group_id, artifact_id, version` fields.

This would probably be good to add to the metadata download / query step.

Reply:
@sdherr what do you think of @jdcasey's comment?